DOC-600 Add cypher and couchbase community outputs from v4.37.0 (#69)
redpanda-data/rp-connect-docs: modules/components/pages/outputs/couchbase.adoc

= couchbase
:type: output

> :status: experimental

Reviewer: If experimental is now considered beta, we can replace this with `:page-beta: true`

Author: Updated as requested.

:categories: ["Integration"]

// © 2024 Redpanda Data Inc.

component_type_dropdown::[]

> Perform operations against a Couchbase database for each message to store or delete data.

Reviewer: ```suggestion
Perform operations on a https://docs.couchbase.com/home/index.html[Couchbase database^] for each incoming message. This connector can store or delete data in Couchbase based on the message's content.
```

Author: Updated as requested.

> Introduced in version 4.37.0.

Reviewer: We should conditionalize this in case we ever want to single-source it into cloud docs.

Author: Updated as requested.

    username: "" # No default (optional)
    password: "" # No default (optional)
    bucket: "" # No default (required)
    id: ${! json("id") } # No default (required)
    content: "" # No default (optional)
    operation: upsert
    max_in_flight: 64
    batching:
      count: 0
      byte_size: 0
      period: ""
      check: ""
```
--
Advanced::
+
--
```yml
# All configuration fields, showing default values
output:
  label: ""
  couchbase:
    url: couchbase://localhost:11210 # No default (required)
    username: "" # No default (optional)
    password: "" # No default (optional)
    bucket: "" # No default (required)
    collection: _default
    transcoder: legacy
    timeout: 15s
    id: ${! json("id") } # No default (required)
    content: "" # No default (optional)
    operation: upsert
    max_in_flight: 64
    batching:
      count: 0
      byte_size: 0
      period: ""
      check: ""
      processors: [] # No default (optional)
```
--
======

== Performance

For improved performance, this output sends multiple messages in parallel. You can tune the maximum number of in-flight messages or message batches, using the `max_in_flight` field.

> This output can also send message batches. You can configure batches at both the input and output level. For more information, see xref:configuration:batching.adoc[Message Batching].

Reviewer: The relationship between 'in-flight messages' and message batches is unclear here.

> You can tune the maximum number of in-flight messages or message batches, using the `max_in_flight` field.

This implies that message batches is a synonym of in-flight messages.

> This output can also send message batches. You can configure batches at both the input and output level. For more information, see xref:configuration:batching.adoc[Message Batching].

Then, this implies that message batches are entirely different. We also don't mention the `batching.*` configuration options here but I think it would be useful.

Author: Updated as requested. As in previous PR, I've raised an issue to sort out this content site wide so it is standardized but also contains contextual information where necessary.

== Fields

> === `url`

Author: Please check field descriptions for accuracy.

The Couchbase connection string.

*Type*: `string`

```yml
# Examples
url: couchbase://localhost:11210
```
=== `username`

A username to authenticate with the Couchbase cluster.

*Type*: `string`

=== `password`

A password to authenticate with the Couchbase cluster.

include::components:partial$secret_warning.adoc[]

*Type*: `string`

=== `bucket`

The name of the Couchbase bucket you want the output to access.

*Type*: `string`

=== `collection`

The name of the specific collection you want the output to write to.

*Type*: `string`

*Default*: `_default`

=== `transcoder`

> Choose from the following Couchbase transcoders to convert the format of the data before it is written.

Reviewer: ```suggestion
Choose from the following Couchbase transcoders to convert the messages' data format before they are written to Couchbase.
```

Author: Updated as requested.
